secrets-plugin/src/test/java/io/airlift/secrets/keystore/TestKeystoreSecretProvider.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.secrets.keystore;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.security.KeyStore;
+
+import static java.io.File.createTempFile;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+
+@TestInstance(PER_CLASS)
+final class TestKeystoreSecretProvider
+{
+    private File keystoreFile;
+
+    private KeystoreSecretProvider secretProvider;
+
+    @BeforeAll
+    public void setup()
+            throws Exception
+    {
+        keystoreFile = createTempFile(System.getProperty("java.io.tmpdir"), "keystore");
```

Comment:
Does airlift have a testing temp directory utility like Trino has?

Reply:
I don't think we have any - Even in most on trino we use `deleteOnExit`

Reply:
Or should we use a different API here ?

Reply:
Thank you @electrum !! Migrated to `TestFile`